### PR TITLE
Add default path params handler for the compilepath util function.

### PR DESCRIPTION
**Pull request by SimoTw**

### CODE REVIEW

File: src/common/utils.ts [hunk 1]
```diff
@@ -521,9 +521,22 @@ export class URIUtils {
...
 
-    public static compilePath(pathTemplate: string, params: object): string {
+    public static compilePath(pathTemplate: string, params: object, defaultPathParams: object): string {
+        /* Add required default key, value pairs for the "toPath" function into a cloned params object. */
+        const withDefaultParams = (pathTemplate: string, params: object, defaultPathParams: object): object => {
+            const requiredKeys = [];
+            const retParams = {...params};
+            pathToRegexp(pathTemplate, requiredKeys);
+            for (const { name } of requiredKeys) {
+                if (!Object.hasOwnProperty.call(params, name)) {
```
Comment (buddhawang) [inline src/common/utils.ts:531]: nit: I'd suggest explicitly `.call(retParams, name)` here.

File: src/common/utils.ts [hunk 2]
```diff
@@ -521,9 +521,22 @@ export class URIUtils {
...
+                    retParams[name] = defaultPathParams.hasOwnProperty(name) ? defaultPathParams[name] : "";
+                }
+            }
+            return retParams;
+        }
         const toPath = compile(pathTemplate, { encode: encodeURIComponent });
-        return toPath(params);
+        const addedParams = withDefaultParams(pathTemplate, params, defaultPathParams);
+        return toPath(addedParams);
```
Comment (buddhawang) [inline src/common/utils.ts:539]: nit: could combine `toPath(withDefaultParams(pathTemplate, params, defaultPathParams));`

**Review by buddhawang — APPROVED** [review]
Approve with nit comments.

File: src/common/utils.ts (hunk 1 above)
Comment (SimoTw) [reply src/common/utils.ts:531]: Change to `retParams.hasOwnProperty(name)` bacause a retParams object must has the hasOwnProperty method.

**Pull request merged**